apps/server/src/modules/learnroom/service/course-sync.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Group, GroupUser } from '@modules/group';
+import { RoleService } from '@modules/role';
+import { Inject, Injectable } from '@nestjs/common';
+import { RoleName } from '@shared/domain/interface';
+import { EntityId } from '@shared/domain/types';
+import {
+	Course,
+	COURSE_REPO,
+	CourseAlreadySynchronizedLoggableException,
+	CourseNotSynchronizedLoggableException,
+	CourseRepo,
+} from '../domain';
+
+@Injectable()
+export class CourseSyncService {
+	constructor(
+		@Inject(COURSE_REPO) private readonly courseRepo: CourseRepo,
+		private readonly roleService: RoleService
+	) {}
+
+	public async startSynchronization(course: Course, group: Group): Promise<void> {
+		if (course.syncedWithGroup) {
+			throw new CourseAlreadySynchronizedLoggableException(course.id);
+		}
+
+		await this.synchronize([course], group);
+	}
+
+	public async stopSynchronization(course: Course): Promise<void> {
+		if (!course.syncedWithGroup) {
+			throw new CourseNotSynchronizedLoggableException(course.id);
+		}
+
+		course.syncedWithGroup = undefined;
+
+		await this.courseRepo.save(course);
+	}
+
+	public async synchronizeCourseWithGroup(group: Group): Promise<void> {
+		const courses: Course[] = await this.courseRepo.findBySyncedGroup(group);
+		await this.synchronize(courses, group);
+	}
+
+	private async synchronize(courses: Course[], group: Group): Promise<void> {
+		if (courses.length) {
+			const [studentRole, teacherRole] = await Promise.all([
+				this.roleService.findByName(RoleName.STUDENT),
+				this.roleService.findByName(RoleName.TEACHER),
+			]);
+
+			const students = group.users.filter((groupUser: GroupUser) => groupUser.roleId === studentRole.id);
+			const teachers = group.users.filter((groupUser: GroupUser) => groupUser.roleId === teacherRole.id);
+
+			const toBeSyncedCourses = courses.map((course) => {
+				course.syncedWithGroup = group.id;
+				course.name = group.name;
```

Comment:
why did you take out the condition for the name update?
1. Existing courses which start a new sync _**should not**_ update their name to the group name.
2. Existing courses that are already synced _**should not**_ update their name to the group name, when the course name is different to the group name (it was changed by the teacher).
3. Existing courses that are already synced _**should**_ update their name to the group name, when the course name is the same as the group name (the name in moin.schule changed and should be synced).